#!/usr/bin/env python
PACKAGE = "hfts_grasp_planner"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add('num_hfts_iterations', int_t, 0, "Maximum number iterations per HFTS layer", 40, 1, 200)
gen.add('num_planning_attempts', int_t, 0, "Maximum number of grasp planning attempts", 10, 1, 300)
gen.add('com_center_weight', double_t, 0, "Weight for grasp quality measure for Robotiq hand.", 0.1, 0.0, 10.0)
gen.add('position_reachability_weight', double_t, 0, "Weight for grasp reachability measure for Robotiq hand.", 0.1, 0.0, 10.0)
gen.add('normal_reachability_weight', double_t, 0, "Weight for grasp reachability measure for Robotiq hand.", 0.1, 0.0, 10.0)
gen.add('reachability_weight', double_t, 0, "Weight for reachability component in objective function", 1.0, 0.0, 1000.0)

exit(gen.generate(PACKAGE, "hfts_planner_node", "HFTSPlanner"))
